firestarter/workflows/build_images/resources/schema.json

```diff
@@ -0,0 +1,108 @@
+  "type": "object",
+  "properties": {
+    "snapshots": {
+      "type": "object",
+      "patternProperties": {
+        "^[a-zA-Z0-9_-]+$": {
+          "type": "object",
+          "properties": {
+            "dockerfile": {
+              "type": "string",
+              "pattern": ".+" 
+            },
+            "auto": {
+              "type": "boolean"
+            },
+            "build_always":{
+                "type": "boolean"
+            },
+            "build_args": {
+              "type": "object",
+              "additionalProperties": true 
+            },
+            "secrets": {
+              "type": "object",
+              "additionalProperties": true 
+            },
+            "extra_registries": {
+              "type": "array",
+              "items": {
+                "type": "object",
+                "properties": {
+                  "name": {
+                    "type": "string"
+                  },
+                  "repository": {
+                    "type": "string"
+                  },
+                  "auth_strategy": {
+                    "type": "string",
+                    "enum": ["azure_oidc", "aws_oidc"]
+                  }
+                },
+                "required": ["name", "repository", "auth_strategy"],
+                "additionalProperties": false 
+              }
+            }
+          },
+          "required": ["dockerfile"], 
+          "additionalProperties": false 
+        }
+      },
+      "minProperties": 1 
+    },
+    "releases": {
+      "type": "object",
+      "patternProperties": {
+        "^[a-zA-Z0-9_-]+$": {
+          "type": "object",
+          "properties": {
+            "dockerfile": {
+              "type": "string",
+              "pattern": ".+" 
+            },
+            "auto": {
+              "type": "boolean"
+            },
+            "build_args": {
+              "type": "object",
+              "additionalProperties": true 
+            },
+            "secrets": {
+              "type": "object",
+              "additionalProperties": true 
+            },
+            "extra_registries": {
+              "type": "array",
+              "items": {
+                "type": "object",
+                "properties": {
+                  "name": {
+                    "type": "string"
+                  },
+                  "repository": {
+                    "type": "string"
+                  },
+                  "auth_strategy": {
+                    "type": "string",
+                    "enum": ["azure_oidc", "aws_oidc"] 
+                  }
+                },
+                "required": ["name", "repository", "auth_strategy"],
+                "additionalProperties": false 
+              }
+            }
+          },
+          "required": ["dockerfile"],  
+          "additionalProperties": false  
+        }
+      },
+      "minProperties": 1  
+    }
+  },
+  "required": ["snapshots", "releases"],  
```

Comment:
Only one of the 2 image types is required, I mean you can define `snapshots` image types, `releases` image types, or both of them